presto-main/src/test/java/com/facebook/presto/operator/BenchmarkReadBlock.java

```diff
@@ -0,0 +1,298 @@
+            if (!data.blockWithNulls.isNullUnchecked(data.positions[i])) {
+                index = newIndex;
+            }
+        }
+        return index;
+    }
+
+    @Benchmark
+    public int randomCopyLongValuesWithDictionary(BenchmarkData data)
+    {
+        int index = 0;
+        for (int i = 0; i < data.longValues.length; i++) {
+            index = setLongUnchecked(data.bytes, index, data.longValues[data.ids[data.positions[i]]]);
+        }
+        return index;
+    }
+
+    @Benchmark
+    public int randomCopyDictionaryBlock(BenchmarkData data)
+    {
+        int index = 0;
+        for (int i = 0; i < data.longValues.length; i++) {
+            index = setLongUnchecked(data.bytes, index, data.dictionaryBlockNoNulls.getLong(data.positions[i]));
+        }
+        return index;
+    }
+
+    @Benchmark
+    public int randomCopyUncheckedDictionaryBlock(BenchmarkData data)
+    {
+        int index = 0;
+        for (int i = 0; i < data.longValues.length; i++) {
+            index = setLongUnchecked(data.bytes, index, data.dictionaryBlockNoNulls.getLongUnchecked(data.positions[i]));
+        }
+        return index;
+    }
+
+    @Benchmark
+    public int randomCopyLongValuesWithDictionaryWithNulls(BenchmarkData data)
+    {
+        int index = 0;
+        for (int i = 0; i < data.longValues.length; i++) {
+            int newIndex = setLongUnchecked(data.bytes, index, data.longValues[data.positions[i]]);
+            if (!data.nulls[data.ids[data.positions[i]]]) {
+                index = newIndex;
+            }
+        }
+        return index;
+    }
+
+    @Benchmark
+    public int randomCopyDictionaryBlockWithNulls(BenchmarkData data)
+    {
+        int index = 0;
+        for (int i = 0; i < data.longValues.length; i++) {
+            int newIndex = setLongUnchecked(data.bytes, index, data.dictionaryBlockWithNulls.getLong(data.positions[i]));
+            if (!data.dictionaryBlockWithNulls.isNull(data.positions[i])) {
+                index = newIndex;
+            }
+        }
+        return index;
+    }
+
+    @Benchmark
+    public int randomCopyUncheckedDictionaryBlockWithNulls(BenchmarkData data)
+    {
+        int index = 0;
+        for (int i = 0; i < data.longValues.length; i++) {
+            int newIndex = setLongUnchecked(data.bytes, index, data.dictionaryBlockWithNulls.getLongUnchecked(data.positions[i]));
+            if (!data.dictionaryBlockWithNulls.isNullUnchecked(data.positions[i])) {
+                index = newIndex;
+            }
+        }
+        return index;
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private static final int POSITIONS_PER_PAGE = 10000;
+
+        private final Random random = new Random(0);
+
+        private final long[] longValues = LongStream.range(0, POSITIONS_PER_PAGE).map(i -> random.nextLong()).toArray();
```

Comment:
nit: I would just use a for-loop to initialize `longValues`, `ids` and ` positions` array.

Reply:
I thought we want to use more succinct code for non-performance-critical code paths. @mbasmanova what do you think?

Reply:
@yingsu00 : Whether to use stream/lambda is one of the never-ending discussion between engineers (other discussions including whether Vim or Emacs is better :P)

For me, I personally prefer only to use stream/lambda when it can be naturally viewed as "element-wise operation", and the underlying operation is generally simple.  Thus I prefer not to use `IntStream/LongStream.range` in general.  See a similar discussion in https://github.com/prestodb/presto/pull/12966#discussion_r301764260 .

But again this is a personal preference, and I am sure there are certain part of Presto codebase I will ask not to use lambda :). So I will leave the judgement to you.